CRAW-239: Clean up author name using Named Entities

abhishekvm opened the pull request:
- Introduced new rawAuthorName field during extraction. This field will be used in juicer web app to extract actual author name using named entities.
- Refactored the code

Commit:
- Introduced new rawAuthorName field during extraction. This field will be used in juicer web app to extract actual author name using named entities.
- Refactored the code

skyshard commented on src/test/java/de/jetwick/snacktory/ArticleTextExtractorTest.java:
```diff
@@ -3046,7 +3046,7 @@ public void testTheRivardReport() throws Exception {
         assertTrue(res.getText(), res.getText().startsWith("As the countdown began on Aug. 5, 2011 at the Kennedy Space Center in Florida,"));
         assertTrue(res.getText(), res.getText().endsWith("Artistic depiction of Juno. Photo courtesy of NASA."));
-        assertEquals("Cherise Rohr-Allegrini", res.getAuthorName());
+        assertEquals("Cherise Rohr Allegrini", res.getAuthorName());
```
This may be too aggressive, some symbols are part of names

skyshard commented on src/main/java/de/jetwick/snacktory/JResult.java:
```diff
@@ -128,8 +129,17 @@ public String getAuthorName() {
-	
+    }
+
+    public String getRawAuthorName() {
```
Need tests for this too

abhishekvm replied:
Added test cases for rawAuthorName https://github.com/skyshard/snacktory/pull/47/files#diff-54669b9ba2e67fc2468e33046ee1e1b8R3534

Commit:
Addressed review comments
- Allowed some special symbols in authorName like dot(.) and hyphen(-)
- Added testcases for getRawAuthorName

Commit:
CRAW-238: Extract author name from facebook profile urls
